scrapy/contrib/downloadermiddleware/robotstxt.py

```diff
@@ -49,8 +49,14 @@ def robot_parser(self, request, spider):
             )
             dfd = self.crawler.engine.download(robotsreq, spider)
             dfd.addCallback(self._parse_robots)
+            dfd.addErrback(self._logerror, robotsreq, spider)
         return self._parsers[netloc]
 
+    def _logerror(self, failure, request, spider):
+        if failure.type is not IgnoreRequest:
+            log.msg(format="Error downloading %%(request)s: %s" % failure.value,
```

Comment:
any reason to not use `%(failure)` and pass `failure=failure.value` keyword argument ?